### PR TITLE
Changes for orchestra.daemon to work with systemd

### PR DESCRIPTION
**Pull request by vasukulkarni**
Use systemd when cluster is setup with ceph-deploy or ceph-ansible,  still in testing..

Signed-off-by: Vasu Kulkarni vasu@redhat.com

### CODE REVIEW

**Comment by vasukulkarni** [comment]
@ktdreyer @athanatos once this works we can replace half of the install task with ceph-deploy and other half with ceph-ansible that will do more customer like testing  for thrash testcases

Currently holding on http://tracker.ceph.com/issues/17050 to be fixed as well, also have to check on what the wait function does once i can reach that stage..

**Comment by athanatos** [comment]
Neat!

**Comment by tchaikov** [comment]
@vasukulkarni  i just closed http://tracker.ceph.com/issues/17050 as "rejected".

**Comment by vasukulkarni** [comment]
@tchaikov thanks, that makes sense, will make changes and rerun.

**Comment by vasukulkarni** [comment]
@zmc this is ready for merge and tested.

File: teuthology/orchestra/daemon.py [hunk 1]
```diff
@@ -41,19 +93,38 @@ def stop(self, timeout=300):
...
+            run.wait([self.proc], timeout=timeout)
+            self.proc = None
         self.log.info('Stopped')
 
+    def start(self, timeout=300):
+        """
+        Start this daemon instance.
+        """
+        if not self.running():
+            self.log.error('Restarting a running daemon')
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: This should probably be a warning

File: teuthology/orchestra/daemon.py [hunk 2]
```diff
@@ -72,6 +143,14 @@ def restart(self, *args, **kwargs):
         :param kwargs: keyword arguments passed to remote.run
         """
         self.log.info('Restarting daemon')
+        if self.use_init:
+            self.log.info("using init to restart")
+            if not self.running():
+                self.log.error('starting a non-running daemon')
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: This should also be a warning I think

File: teuthology/orchestra/daemon.py [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+stop_rgw_cmd = 'sudo systemctl stop ceph-radosgw@rgw.'
+restart_rgw_cmd = 'sudo systemctl restart ceph-radosgw@rgw.'
+
 
 class DaemonState(object):
     """
     Daemon State.  A daemon exists for each instance of each role.
     """
-    def __init__(self, remote, role, id_, *command_args, **command_kwargs):
+    def __init__(self, remote, role, id_, use_init=False, *command_args, **command_kwargs):
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: What is `use_init` ? Should it be `use_sytemd` ?

File: teuthology/orchestra/daemon.py [hunk 4]
```diff
@@ -88,7 +167,13 @@ def restart_with_args(self, extra_args):
 
         :param extra_args: Extra keyword arguments to be added.
         """
-        self.log.info('Restarting daemon')
+        self.log.info('Restarting daemon with args')
+        if self.use_init:
+            self.log.info("restart with args not supported in init")
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: Perhaps this should be a warning - and it should mention systemd, no?

File: teuthology/orchestra/daemon.py [hunk 5]
```diff
@@ -108,6 +193,16 @@ def signal(self, sig, silent=False):
 
         :param sig: signal to send
         """
+        if self.use_init:
+            self.log.info("using init to send signal")
+            self.log.info("WARNING init may restart after kill signal")
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: - More vague usage of the term "init"
- `log.info()` and 'WARNING' ?
- What exactly does "init may restart" mean?

File: teuthology/orchestra/daemon.py [hunk 6]
```diff
@@ -117,18 +212,35 @@ def running(self):
         Are we running?
         :return: True if remote run command value is set, False otherwise.
         """
+        if self.use_init:
+            self.log.info("using init to send signal")
+            self.log.info("WARNING init may restart after kill signal")
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: See above comment

File: teuthology/orchestra/daemon.py [hunk 7]
```diff
@@ -117,18 +212,35 @@ def running(self):
...
+            else:
+                return None
         return self.proc is not None
 
     def reset(self):
         """
         clear remote run command value.
         """
+        if self.use_init:
+            self.log.info("reset not supported with init")
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: Does this really need to log anything?

File: teuthology/orchestra/daemon.py [hunk 8]
```diff
@@ -117,18 +212,35 @@ def running(self):
...
+            return
         self.proc = None
 
     def wait_for_exit(self):
         """
         clear remote run command value after waiting for exit.
         """
+        if self.use_init:
+            self.log.info("wait_for_exit not supported with init")
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: Is logging necessary here?

File: teuthology/orchestra/daemon.py (hunk 4 above)
Comment (zmc) [inline teuthology/orchestra/daemon.py]: or `raise NotImplementedError` ?

File: teuthology/orchestra/daemon.py [hunk 9]
```diff
@@ -26,6 +43,41 @@ def __init__(self, remote, role, id_, *command_args, **command_kwargs):
...
         self.id_ = id_
+        self.use_init = use_init
+        if self.use_init:
+            _, role = role.split('.')
+            self.proc_name = 'ceph-{role}'.format(role=role)
+            if (role == 'mon') or (role == 'mds') or (role == 'rgw'):
+                self.id = remote.shortname
+            else:
+                self.id = id_
+            self.proc_regex = '"' + self.proc_name + '.*--id ' + self.id + '"'
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: Let's split out the pid-getting functionality into a separate method

File: teuthology/orchestra/daemon.py [hunk 10]
```diff
@@ -26,6 +43,41 @@ def __init__(self, remote, role, id_, *command_args, **command_kwargs):
...
+            self.list_proc_id = ['ps', '-ef',
+                                 run.Raw('|'),
+                                 'grep',
+                                 run.Raw(self.proc_regex),
+                                 run.Raw('|'),
+                                 'grep', '-v',
+                                 'grep', run.Raw('|'),
+                                 'awk',
+                                 run.Raw("{'print $2'}")]
```
Comment (zmc) [inline teuthology/orchestra/daemon.py]: Instead of the above, could we use `systemctl status` ?

**Comment by zmc** [comment]
@vasukulkarni, any ETA on this one?

**Comment by vasukulkarni** [comment]
internal task pr should be ready for monday, soon after that I will pick this one.

**Comment by zmc** [comment]
@vasukulkarni and I agreed that I should help out with this PR. I'll rebase it and make changes as needed. In the meantime I will make this a DNM as it's not been tested.

**Comment by zmc** [comment]
@vasukulkarni I've cleaned up the branch. Note:

- It is still enabled by default on machines that use systemd. I need to think of how to make it configurable in a sane way
- It will probably not work just yet (needs testing), so I'm less immediately concerned about its default state
- It needs the changes in https://github.com/zmc/ceph/tree/wip-master-init to run

**Comment by vasukulkarni** [comment]
@zmc awesome thanks!  

for 1) we can try to use config at higher level eg: overrides: disable-daemon-helper: true or something so that its easier to run suites with overrides

**Comment by vasukulkarni** [comment]
These are the first 10 tests that I ran with these changes and I have checked logs for few and dont see any  issues 💯 ,  if/when we enable the option to configure it from yaml we could starting enabling it per suite basis.

http://pulpito.ceph.com/vasu-2017-09-29_22:02:40-smoke-luminous-distro-basic-ovh/

```
2017-09-29T22:23:17.464 INFO:tasks.ceph:Starting mon daemons in cluster ceph...
2017-09-29T22:23:17.464 INFO:teuthology.orchestra.run.ovh081:Running: 'which systemctl'
2017-09-29T22:23:17.620 INFO:teuthology.orchestra.run.ovh081.stdout:/bin/systemctl
2017-09-29T22:23:17.620 INFO:tasks.ceph.mon.a:Restarting daemon using systemd
2017-09-29T22:23:17.620 INFO:teuthology.orchestra.run.ovh081:Running: 'ps -ef | grep "ceph-mon.*--id a" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:17.806 INFO:tasks.ceph.mon.a:starting a non-running daemon
2017-09-29T22:23:17.806 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl start ceph-mon@a'
2017-09-29T22:23:17.993 INFO:teuthology.orchestra.run.ovh013:Running: 'which systemctl'
2017-09-29T22:23:18.096 INFO:teuthology.orchestra.run.ovh013.stdout:/bin/systemctl
2017-09-29T22:23:18.097 INFO:tasks.ceph.mon.b:Restarting daemon using systemd
2017-09-29T22:23:18.097 INFO:teuthology.orchestra.run.ovh013:Running: 'ps -ef | grep "ceph-mon.*--id b" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:18.255 INFO:tasks.ceph.mon.b:starting a non-running daemon
2017-09-29T22:23:18.256 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl start ceph-mon@b'
2017-09-29T22:23:18.444 INFO:tasks.ceph.mon.c:Restarting daemon using systemd
2017-09-29T22:23:18.445 INFO:teuthology.orchestra.run.ovh013:Running: 'ps -ef | grep "ceph-mon.*--id c" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:18.604 INFO:tasks.ceph.mon.c:starting a non-running daemon
2017-09-29T22:23:18.605 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl start ceph-mon@c'
2017-09-29T22:23:18.773 INFO:tasks.ceph:Starting mgr daemons in cluster ceph...
2017-09-29T22:23:18.773 INFO:tasks.ceph.mgr.x:Restarting daemon using systemd
2017-09-29T22:23:18.773 INFO:teuthology.orchestra.run.ovh081:Running: 'ps -ef | grep "ceph-mgr.*--id x" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:18.900 INFO:tasks.ceph.mgr.x:starting a non-running daemon
2017-09-29T22:23:18.900 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl start ceph-mgr@x'
2017-09-29T22:23:19.068 INFO:tasks.ceph.mgr.y:Restarting daemon using systemd
2017-09-29T22:23:19.068 INFO:teuthology.orchestra.run.ovh013:Running: 'ps -ef | grep "ceph-mgr.*--id y" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:19.182 INFO:tasks.ceph.mgr.y:starting a non-running daemon
2017-09-29T22:23:19.183 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl start ceph-mgr@y'
2017-09-29T22:23:19.352 INFO:tasks.ceph:Setting crush tunables to default
2017-09-29T22:23:19.353 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo ceph --cluster ceph osd crush tunables default'
2017-09-29T22:23:28.929 INFO:teuthology.orchestra.run.ovh081.stderr:adjusted tunables profile to default
2017-09-29T22:23:28.953 INFO:tasks.ceph:Starting osd daemons in cluster ceph...
2017-09-29T22:23:28.953 INFO:teuthology.orchestra.run.ovh081:Running: "python -c 'import os; import tempfile; import sys;(fd,fname) = tempfile.mkstemp();os.close(fd);sys.stdout.write(fname.rstrip());sys.stdout.flush()'"
2017-09-29T22:23:29.128 INFO:teuthology.orchestra.run.ovh081.stdout:/tmp/tmpELMXtd
2017-09-29T22:23:29.128 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo cp /var/lib/ceph/osd/ceph-0/fsid /tmp/tmpELMXtd'
2017-09-29T22:23:29.291 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo chmod 0666 /tmp/tmpELMXtd'
2017-09-29T22:23:29.717 DEBUG:teuthology.orchestra.remote:ovh081:/tmp/tmpELMXtd is 37B
2017-09-29T22:23:30.094 INFO:teuthology.orchestra.run.ovh081:Running: 'rm -fr /tmp/tmpELMXtd'
2017-09-29T22:23:30.197 INFO:teuthology.orchestra.run.ovh081:Running: "python -c 'import os; import tempfile; import sys;(fd,fname) = tempfile.mkstemp();os.close(fd);sys.stdout.write(fname.rstrip());sys.stdout.flush()'"
2017-09-29T22:23:30.354 INFO:teuthology.orchestra.run.ovh081.stdout:/tmp/tmpjm5VFo
2017-09-29T22:23:30.355 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo cp /var/lib/ceph/osd/ceph-1/fsid /tmp/tmpjm5VFo'
2017-09-29T22:23:30.500 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo chmod 0666 /tmp/tmpjm5VFo'
2017-09-29T22:23:30.926 DEBUG:teuthology.orchestra.remote:ovh081:/tmp/tmpjm5VFo is 37B
2017-09-29T22:23:31.305 INFO:teuthology.orchestra.run.ovh081:Running: 'rm -fr /tmp/tmpjm5VFo'
2017-09-29T22:23:31.412 INFO:teuthology.orchestra.run.ovh013:Running: "python -c 'import os; import tempfile; import sys;(fd,fname) = tempfile.mkstemp();os.close(fd);sys.stdout.write(fname.rstrip());sys.stdout.flush()'"
2017-09-29T22:23:31.540 INFO:teuthology.orchestra.run.ovh013.stdout:/tmp/tmpRkW24A
2017-09-29T22:23:31.541 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo cp /var/lib/ceph/osd/ceph-2/fsid /tmp/tmpRkW24A'
2017-09-29T22:23:31.689 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo chmod 0666 /tmp/tmpRkW24A'
2017-09-29T22:23:32.110 DEBUG:teuthology.orchestra.remote:ovh013:/tmp/tmpRkW24A is 37B
2017-09-29T22:23:32.451 INFO:teuthology.orchestra.run.ovh013:Running: 'rm -fr /tmp/tmpRkW24A'
2017-09-29T22:23:32.555 INFO:teuthology.orchestra.run.ovh013:Running: "python -c 'import os; import tempfile; import sys;(fd,fname) = tempfile.mkstemp();os.close(fd);sys.stdout.write(fname.rstrip());sys.stdout.flush()'"
2017-09-29T22:23:32.679 INFO:teuthology.orchestra.run.ovh013.stdout:/tmp/tmpyKCeG6
2017-09-29T22:23:32.680 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo cp /var/lib/ceph/osd/ceph-3/fsid /tmp/tmpyKCeG6'
2017-09-29T22:23:32.827 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo chmod 0666 /tmp/tmpyKCeG6'
2017-09-29T22:23:33.256 DEBUG:teuthology.orchestra.remote:ovh013:/tmp/tmpyKCeG6 is 37B
2017-09-29T22:23:33.596 INFO:teuthology.orchestra.run.ovh013:Running: 'rm -fr /tmp/tmpyKCeG6'
2017-09-29T22:23:33.698 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo ceph --cluster ceph osd new 7acbb9d6-5c31-4e38-b263-d10e47c81467 0'
2017-09-29T22:23:34.162 INFO:teuthology.orchestra.run.ovh013.stdout:0
2017-09-29T22:23:34.181 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo ceph --cluster ceph osd new cea06024-b985-4071-8f31-6255ec64aeb1 1'
2017-09-29T22:23:34.597 INFO:teuthology.orchestra.run.ovh013.stdout:1
2017-09-29T22:23:34.618 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo ceph --cluster ceph osd new 908f5340-b3c8-4077-afd7-fb39d049848e 2'
2017-09-29T22:23:35.025 INFO:teuthology.orchestra.run.ovh013.stdout:2
2017-09-29T22:23:35.043 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo ceph --cluster ceph osd new 699ab681-3549-4950-8111-733baf792edd 3'
2017-09-29T22:23:35.429 INFO:teuthology.orchestra.run.ovh013.stdout:3
2017-09-29T22:23:35.443 INFO:tasks.ceph.osd.0:Restarting daemon using systemd
2017-09-29T22:23:35.443 INFO:teuthology.orchestra.run.ovh081:Running: 'ps -ef | grep "ceph-osd.*--id 0" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:35.561 INFO:tasks.ceph.osd.0:starting a non-running daemon
2017-09-29T22:23:35.561 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl start ceph-osd@0'
2017-09-29T22:23:35.776 INFO:tasks.ceph.osd.1:Restarting daemon using systemd
2017-09-29T22:23:35.776 INFO:teuthology.orchestra.run.ovh081:Running: 'ps -ef | grep "ceph-osd.*--id 1" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:35.924 INFO:tasks.ceph.osd.1:starting a non-running daemon
2017-09-29T22:23:35.924 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl start ceph-osd@1'
2017-09-29T22:23:36.142 INFO:tasks.ceph.osd.2:Restarting daemon using systemd
2017-09-29T22:23:36.142 INFO:teuthology.orchestra.run.ovh013:Running: 'ps -ef | grep "ceph-osd.*--id 2" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:36.232 INFO:tasks.ceph.osd.2:starting a non-running daemon
2017-09-29T22:23:36.232 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl start ceph-osd@2'
2017-09-29T22:23:36.445 INFO:tasks.ceph.osd.3:Restarting daemon using systemd
2017-09-29T22:23:36.445 INFO:teuthology.orchestra.run.ovh013:Running: 'ps -ef | grep "ceph-osd.*--id 3" | grep -v grep | awk {\'print $2\'}'
2017-09-29T22:23:36.603 INFO:tasks.ceph.osd.3:starting a non-running daemon
2017-09-29T22:23:36.603 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl start ceph-osd@3'
2017-09-29T22:23:36.829 INFO:tasks.ceph:Waiting for OSDs to come up
2017-09-29T22:23:36.830 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl show ceph-osd@1 | grep -i state'
2017-09-29T22:23:36.944 INFO:teuthology.orchestra.run.ovh081.stdout:LoadState=loaded
2017-09-29T22:23:36.944 INFO:teuthology.orchestra.run.ovh081.stdout:ActiveState=active
2017-09-29T22:23:36.944 INFO:teuthology.orchestra.run.ovh081.stdout:SubState=running
2017-09-29T22:23:36.945 INFO:teuthology.orchestra.run.ovh081.stdout:UnitFileState=disabled
2017-09-29T22:23:36.945 INFO:teuthology.orchestra.run.ovh081.stdout:StateChangeTimestamp=Fri 2017-09-29 22:23:36 UTC
2017-09-29T22:23:36.945 INFO:teuthology.orchestra.run.ovh081.stdout:StateChangeTimestampMonotonic=687976204
2017-09-29T22:23:36.945 INFO:teuthology.orchestra.run.ovh081:Running: 'sudo systemctl show ceph-osd@0 | grep -i state'
2017-09-29T22:23:37.095 INFO:teuthology.orchestra.run.ovh081.stdout:LoadState=loaded
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh081.stdout:ActiveState=active
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh081.stdout:SubState=running
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh081.stdout:UnitFileState=disabled
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh081.stdout:StateChangeTimestamp=Fri 2017-09-29 22:23:35 UTC
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh081.stdout:StateChangeTimestampMonotonic=687612366
2017-09-29T22:23:37.096 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl show ceph-osd@3 | grep -i state'
2017-09-29T22:23:37.215 INFO:teuthology.orchestra.run.ovh013.stdout:LoadState=loaded
2017-09-29T22:23:37.215 INFO:teuthology.orchestra.run.ovh013.stdout:ActiveState=active
2017-09-29T22:23:37.215 INFO:teuthology.orchestra.run.ovh013.stdout:SubState=running
2017-09-29T22:23:37.216 INFO:teuthology.orchestra.run.ovh013.stdout:UnitFileState=disabled
2017-09-29T22:23:37.216 INFO:teuthology.orchestra.run.ovh013.stdout:StateChangeTimestamp=Fri 2017-09-29 22:23:36 UTC
2017-09-29T22:23:37.216 INFO:teuthology.orchestra.run.ovh013.stdout:StateChangeTimestampMonotonic=683024266
2017-09-29T22:23:37.216 INFO:teuthology.orchestra.run.ovh013:Running: 'sudo systemctl show ceph-osd@2 | grep -i state'
2017-09-29T22:23:37.370 INFO:teuthology.orchestra.run.ovh013.stdout:LoadState=loaded
2017-09-29T22:23:37.371 INFO:teuthology.orchestra.run.ovh013.stdout:ActiveState=active
2017-09-29T22:23:37.371 INFO:teuthology.orchestra.run.ovh013.stdout:SubState=running
2017-09-29T22:23:37.371 INFO:teuthology.orchestra.run.ovh013.stdout:UnitFileState=disabled
2017-09-29T22:23:37.371 INFO:teuthology.orchestra.run.ovh013.stdout:StateChangeTimestamp=Fri 2017-09-29 22:23:36 UTC
2017-09-29T22:23:37.371 INFO:teuthology.orchestra.run.ovh013.stdout:StateChangeTimestampMonotonic=682640410
2017-09-29T22:23:37.372 INFO:teuthology.orchestra.run.ovh081:Running: 'adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage ceph --cluster ceph osd dump --format=json'
```

**Comment by vasukulkarni** [comment]
I will also schedule a full smoke suite today

**Comment by vasukulkarni** [comment]
I think we are probably missing some setting, need to check with @jdurgin  on what needs to be done for this to pass

http://qa-proxy.ceph.com/teuthology/vasu-2017-10-03_18:00:35-smoke-luminous-distro-basic-ovh/1699376/teuthology.log
```
2017-10-03T18:32:34.026 INFO:teuthology.orchestra.run.ovh028:Running: 'sudo adjust-ulimits ceph-coverage /home/ubuntu/cephtest/archive/coverage timeout 120 ceph --cluster ceph osd last-stat-seq osd.0'
2017-10-03T18:32:34.219 INFO:teuthology.orchestra.run.ovh028.stderr:2017-10-03 18:32:34.193614 7f45c6cfd700 -1 WARNING: all dangerous and experimental features are enabled.
2017-10-03T18:32:34.241 INFO:teuthology.orchestra.run.ovh028.stderr:2017-10-03 18:32:34.215396 7f45c6cfd700 -1 WARNING: all dangerous and experimental features are enabled.
2017-10-03T18:32:34.434 INFO:teuthology.orchestra.run.ovh028.stdout:0
2017-10-03T18:32:34.434 INFO:tasks.ceph.ceph_manager.ceph:need seq 55834574867
 got 0
 for osd.0
```

b) http://tracker.ceph.com/issues/21671

**Comment by vasukulkarni** [comment]
I will fix the current cluster configuration to properly test with systemd without overloads and rerun, not really great run with current cluster config here http://pulpito.ceph.com/vasu-2017-10-03_18:00:35-smoke-luminous-distro-basic-ovh/

**Comment by vasukulkarni** [comment]
@zmc  this is ready to be merged! 

Tested here: http://pulpito.ceph.com/vasu-2017-10-14_17:53:49-smoke-luminous-distro-basic-ovh/ 
the cephfs failures are due to distro kernel

I also used the qa branch at : https://github.com/zmc/ceph/tree/wip-master-init , so we should be able to merge both to master.

**Comment by vasukulkarni** [comment]
A better run http://pulpito.ceph.com/vasu-2017-10-17_04:01:21-smoke-luminous-testing-basic-ovh/

**Comment by zmc** [comment]
@vasukulkarni is there a run that mirrors one of those but uses master branches?

**Comment by vasukulkarni** [comment]
@zmc scheduled here now, just used stable branch before http://pulpito.ceph.com/vasu-2017-10-17_22:00:28-smoke-master-testing-basic-ovh/

**Comment by zmc** [comment]
@vasukulkarni that's not using master for teuthology :-/

**Comment by vasukulkarni** [comment]
@zmc its not clear what you are asking in comment,  I dont understand why 'master' branch of teuthology needs to be tested?

**Comment by vasukulkarni** [comment]
@zmc ok if you need for comparision here from the nightly runs http://pulpito.ceph.com/teuthology-2017-10-17_05:02:02-smoke-master-testing-basic-ovh/ 

Additional 2 rgw is due to own qa branch that does not have equivalent branch in s3tests git repo.

**Comment by zmc** [comment]
@vasukulkarni ah, that's definitely helpful, thanks!

**Comment by vasukulkarni** [comment]
Can t wait for this to be merged! , Please merge when you are done reviewing, I can merge the qa part.

**Comment by zmc** [comment]
@vasukulkarni because of the way we've done this PR, it won't let you formally review. So if you could give a thumbs up or something, I can review and then we can merge.

**Comment by vasukulkarni** [comment]
:+1:  :+1:  LGTM  and is ready for merge along with qa branch at https://github.com/zmc/ceph/tree/wip-master-init

**Comment by zmc** [comment]
@vasukulkarni - just filed the ceph PR

**Comment by vasukulkarni** [comment]
@zmc thanks, will merge both once the qa is green.

**Pull request merged**